GDJS/GDJS/Events/CodeGeneration/BehaviorCodeGenerator.cpp

```diff
@@ -148,6 +189,30 @@ CODE_NAMESPACE.RUNTIME_BEHAVIOR_CLASSNAME = class RUNTIME_BEHAVIOR_CLASSNAME ext
   PROPERTIES_CODE
 }
 
+/**
+ * Shared data generated from BEHAVIOR_FULL_NAME
+ */
+CODE_NAMESPACE.RUNTIME_BEHAVIOR_CLASSNAME.SharedData = class RUNTIME_BEHAVIOR_CLASSNAMESharedData {
+  constructor(sharedData) {
+    INITIALIZE_SHARED_PROPERTIES_CODE
+  }
+  
+  // Shared properties:
+  SHARED_PROPERTIES_CODE
+}
+
+CODE_NAMESPACE.RUNTIME_BEHAVIOR_CLASSNAME.getSharedData = function(instanceContainer, behaviorName) {
+  if (!instanceContainer.RUNTIME_BEHAVIOR_CLASSNAMESharedData) {
+    const initialData = instanceContainer.getInitialSharedDataForBehavior(
+      behaviorName
+    );
+    instanceContainer.RUNTIME_BEHAVIOR_CLASSNAMESharedData = new CODE_NAMESPACE.RUNTIME_BEHAVIOR_CLASSNAME.SharedData(
+      initialData
+    );
+  }
+  return instanceContainer.RUNTIME_BEHAVIOR_CLASSNAMESharedData;
```

Comment:
I wonder if this would be useful:

```suggestion
  if (!instanceContainer._RUNTIME_BEHAVIOR_CLASSNAMESharedData) {
    const initialData = instanceContainer.getInitialSharedDataForBehavior(
      behaviorName
    );
    instanceContainer._RUNTIME_BEHAVIOR_CLASSNAMESharedData = new CODE_NAMESPACE.RUNTIME_BEHAVIOR_CLASSNAME.SharedData(
      initialData
    );
  }
  return instanceContainer._RUNTIME_BEHAVIOR_CLASSNAMESharedData;
```

to make it clear this is not to be used directly (instead, `CODE_NAMESPACE.RUNTIME_BEHAVIOR_CLASSNAME.getSharedData` should be used).

Reply:
Oh, yes, I'm making it `_EXTENSION_NAME_RUNTIME_RUNTIME_BEHAVIOR_CLASSNAMESharedData` to avoid name collision.